tests/queue/queue_properties.cpp

```diff
@@ -20,56 +20,237 @@
 *******************************************************************************/
 
 #include "../common/common.h"
+#include "../common/disabled_for_test_case.h"
 
 #define TEST_NAME queue_properties
 
-namespace TEST_NAMESPACE {
+namespace queue_properties {
 
 using namespace sycl_cts;
 
-/** tests the properties for sycl::queue
- */
-class TEST_NAME : public util::test_base {
- public:
-  /** return information about this test
-   */
-  virtual void get_info(test_base::info &out) const override {
-    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
-  }
-
-  /** execute this test
-  */
-  virtual void run(util::logger &log) override {
-    {
-      auto queue = util::get_cts_object::queue();
-
-      /** check property::queue::enable_profiling
-      */
-      {
-        sycl::queue queue(
-            util::get_cts_object::device(),
-            sycl::property_list{
-                sycl::property::queue::enable_profiling()});
-
-        if (!queue
-                 .has_property<sycl::property::queue::enable_profiling>()) {
-          FAIL(log,
-               "queue with enable_profiling property was not constructed "
-               "correctly");
+void check_in_order_prop(const sycl::queue &queue) {
+  CHECK(queue.is_in_order());
+
+  REQUIRE(queue.has_property<sycl::property::queue::in_order>());
+
+  auto prop = queue.get_property<sycl::property::queue::in_order>();
+  check_return_type<sycl::property::queue::in_order>(
+      prop,
+      "sycl::queue::has_property<sycl::property::queue::"
+      "in_order>()");
+}
+
+void check_in_order_functionality(sycl::queue &queue) {
+  bool *data_changed = sycl::malloc_device<bool>(1, queue);
+  constexpr size_t buffer_size = 10;
+  int loop_array[buffer_size];
+  bool result = false;
+  {
+    sycl::buffer<bool, 1> res_buf(&result, sycl::range<1>(1));
+    sycl::buffer<int, 1> loop_buf(loop_array, sycl::range(buffer_size));
+    // to garantee that data_changed initialized as false before two tested
+    // commands are submitted
+    queue
+        .submit([&](sycl::handler &cgh) {
+          cgh.single_task<class kernel1>([=] { *data_changed = false; });
```

Comment:
You need to make forward declarations for kernel typenames to fix the compilation with ComputeCPP.

Reply:
Thank you, done